File: eio2023-ev/2023-12-10-ev/numbrid/solution/sol2.cpp
// sorteerib soovitud numbrite jada ja leiab selles "augud"

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	int n;
	cin >> n;
	vector<int> soovid(n);
	for (auto & soov : soovid) {
		cin >> soov;
	}

	sort(soovid.begin(), soovid.end());

	vector<int> vastus;
	for (int i = 1; i < n; ++i) {
		for (int number = soovid[i - 1] + 1; number < soovid[i]; ++number) {
			vastus.push_back(number);
		}
	}

	cout << vastus.size() << '\n';
	for (auto number : vastus) {
		cout << number << ' ';
	}
	cout << '\n';
}

File: eio2023-ev/2023-12-10-ev/numbrid/solution/sol2.py
# sorteerib soovitud numbrite jada ja leiab selles "augud"

n = int(input())
soovid = [int(s) for s in input().split()]

soovid.sort()

vastus = []
for i in range(1, n):
	for number in range(soovid[i - 1] + 1, soovid[i]):
		vastus.append(number)

print(len(vastus))
print(*vastus)
